dotCMS/src/main/java/com/dotcms/contenttype/business/BaseTypeToContentTypeStrategyResolver.java

```diff
@@ -0,0 +1,120 @@
+package com.dotcms.contenttype.business;
+
+import com.dotcms.contenttype.model.type.BaseContentType;
+import com.dotcms.contenttype.model.type.ContentType;
+import com.dotcms.contenttype.model.type.DotAssetContentType;
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.google.common.collect.ImmutableMap;
+import com.liferay.portal.model.User;
+
+import java.io.File;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * Subscribe Strategies to resolve from base types a content type
+ * and get the strategy for a set of arguments if applies
+ * @author jsanca
+ */
+public class BaseTypeToContentTypeStrategyResolver {
+
+    private volatile Map<BaseContentType, BaseTypeToContentTypeStrategy> strategiesMap = this.getDefaultStrategies();
+
+    private  Map<BaseContentType, BaseTypeToContentTypeStrategy> getDefaultStrategies() {
+
+        final ImmutableMap.Builder<BaseContentType, BaseTypeToContentTypeStrategy> builder =
+                new ImmutableMap.Builder<>();
+
+        builder.put(BaseContentType.DOTASSET, new DotAssetBaseTypeToContentTypeStrategyImpl());
+
+        return builder.build();
+    }
+
+
+    public synchronized void subscribe (final BaseContentType baseContentType, final BaseTypeToContentTypeStrategy strategy) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use block level rather than method level synchronization](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5028307)